### PR TITLE
[WIP] Scoll to selection tweaks, possible fix for #1032

### PR DESCRIPTION
**Pull request by skogsmaskin**
As the editor may be inside a scrollcontainer with it's own scrolling, the ``scrollToSelection`` should look for this first and scroll there,  otherwise deal with ``window``.

This is a proposal for how this can be checked and dealt with accordingly.

Se more about this issue in https://github.com/ianstormtaylor/slate/issues/1032

Before:

Before

After:

After

### CODE REVIEW

**Comment by skogsmaskin** [comment]
In an ideal world it should maybe also act on every parental scoll container, not just the first.

**Comment by ianstormtaylor** [comment]
Hey @skogsmaskin I'm sorry I never commented on this. I like this idea a lot. Can you make it recursive to support nested parent containers, instead of just one? Let me know! Thanks.

**Comment by stoarca** [comment]
Can this be accepted as is, with another PR made for the recursive case? I think this commit solves the problem for 80% of use cases, and the eventual upgrade for the recursive case won't regress or cause any breaking changes.

File: packages/slate-react/src/utils/scroll-to-selection.js [hunk 1]
```diff
@@ -11,23 +33,45 @@ import isBackward from 'selection-is-backward'
 function scrollToSelection(selection) {
   if (!selection.anchorNode) return
 
-  const window = getWindow(selection.anchorNode)
+  const scrollContainer = findScrollContainer(selection.anchorNode)
+  const scrollContainerIsWindow = scrollContainer == scrollContainer.window
```
Comment (ianstormtaylor) [inline packages/slate-react/src/utils/scroll-to-selection.js]: Will `scrollContainer == scrollContainer.window` ever be true? I'm not sure what the `.window` property is?

File: packages/slate-react/src/utils/scroll-to-selection.js [hunk 2]
```diff
@@ -11,23 +50,47 @@ import isBackward from 'selection-is-backward'
 function scrollToSelection(selection) {
   if (!selection.anchorNode) return
 
-  const window = getWindow(selection.anchorNode)
+  const scroller = findScrollContainer(selection.anchorNode)
+  const isWindow = scroller == scroller.window
```
[the remark this answers is not in the document]
Comment (isubastiCadmus) [reply packages/slate-react/src/utils/scroll-to-selection.js]: shouldn't this be scroller == window to check if the scrollContainer is actually window?

[the remark this answers is not in the document]
Comment (isubastiCadmus) [reply packages/slate-react/src/utils/scroll-to-selection.js]: actually the current implementation should be the correct one.
here is jQuery isWindow implementation https://github.com/jquery/jquery/blob/c18d608537d8945de6c5855a9475657177fa74ac/src/var/isWindow.js

**Pull request merged**